server/server.go

```diff
@@ -405,12 +405,18 @@ func (s *Server) Close() {
 
 	log.Info("closing server")
 
-	s.stopServerLoop()
+	if s.httpClient != nil {
+		s.httpClient.CloseIdleConnections()
+	}
 
 	if s.client != nil {
-		s.client.Close()
+		if err := s.client.Close(); err != nil {
+			log.Error("close etcd client meet error", zap.Error(err))
+		}
 	}
 
+	s.stopServerLoop()
```

Comment:
why move this line?

Reply:
Seems to cause some problems, has been reverted